command/agent/config.go

```diff
@@ -307,9 +307,10 @@ type ClientConfig struct {
 // rendering
 type ClientTemplateConfig struct {
 
-	// FunctionBlacklist disables functions in consul-template that
+	// FunctionDenylist disables functions in consul-template that
 	// are unsafe because they expose information from the client host.
-	FunctionBlacklist []string `hcl:"function_blacklist"`
+	// COMPAT(0.13) consul-template uses inclusive language from v0.25.0 - function_blacklist is kept for compatibility
+	FunctionDenylist []string `hcl:"function_denylist" hcl:"function_blacklist"`
```

Comment:
I built Nomad from this PR. The `function_denylist` configuration does work to block the function.

But to verify backwards compatibility I tried to use the older `function_blacklist` syntax in the configuration: it's accepted by the client config parsing but it doesn't end up blocking access to the function anymore. I'm not certain this syntax for go field tags actually works? A couple things you could try here:

* Try with the fields tags swapped
* Add a new field for `function_denylist` and have it accept the value of `function_blacklist` if not set.

Reply:
Ok, it looks like the `template` stanza doesn't validate its keys fully (I saw this on another sub-section recently so I think it's a general bug in our config parser), so it'll accept nonsense fields. For example, this client block works just fine:

```hcl
client {
  enabled = true

  template {
    whatevs = true
    function_denylist = ["env"]
  }
}
```

So in this branch, the `function_blacklist` is being accepted by an overly generous config parser.

Reply:
I've opened https://github.com/hashicorp/nomad/issues/9001 about the overly-generous config parser.

Reply:
This was very interesting. It worked if I left it as default (["plugin"]), but it did NOT fail if `function_blacklist = ["plugin"]`. It did however work to disable the denylist with both `function_blacklist = [""]` and function_denylist = [""]

Either way, changed it so both variables are available.

test job:
```
job "test" {
  datacenters = ["dc1"]
  group "test" {
    task "example" {
      driver = "raw_exec"

      config {
        command = "/bin/sleep"
        args    = ["100"]
      }
      template {
        data        = "{{ plugin \"cat\" \"/etc/hosts\" }}"
        destination = "local/file.txt"
      }
    }
  }
}
```

👍 task **fails** without any config

👍 task **fails** with config ( equals default )
```
client {
  enabled = true
  template {
    function_denylist = ["plugin"]
  }
}
```

👍 task **fails** with config ( equals deafult )
```
client {
  enabled = true
  template {
    function_blacklist = ["plugin"]
  }
}
```
👍 task **works** with config
```
client {
  enabled = true
  template {
    function_blacklist = [""]
  }
}
```
👍 task **works** with config
```
client {
  enabled = true
  template {
    function_denylist = [""]
  }
}
```